includes/documents/class-wcpdf-invoice.php

```diff
@@ -17,16 +19,31 @@
  */
 
 class Invoice extends Order_Document_Methods {
+	
+	public $type;
+	public $title;
+	public $icon;
+	public $lock_name;
+	public $lock_context;
+	public $lock_time;
+	public $lock_retries;
+	
 	/**
 	 * Init/load the order object.
 	 *
 	 * @param  int|object|WC_Order $order Order to init.
 	 */
 	public function __construct( $order = 0 ) {
 		// set properties
-		$this->type		= 'invoice';
-		$this->title	= __( 'Invoice', 'woocommerce-pdf-invoices-packing-slips' );
-		$this->icon		= WPO_WCPDF()->plugin_url() . "/assets/images/invoice.svg";
+		$this->type		    = 'invoice';
+		$this->title	    = __( 'Invoice', 'woocommerce-pdf-invoices-packing-slips' );
+		$this->icon		    = WPO_WCPDF()->plugin_url() . "/assets/images/invoice.svg";
+		
+		// semaphore
+		$this->lock_name    = "wpo_wcpdf_{$this->slug}_semaphore_lock";
+		$this->lock_context = array( 'source' => "wpo-wcpdf-{$this->type}-semaphore" );
+		$this->lock_time    = apply_filters( 'wpo_wcpdf_semaphore_lock_time', 300 );
```

Comment:
Isn't `300` seconds long for a lock? If anything goes wrong in the invoice number creation no invoice number can be created for 5 minutes.

Reply:
Which number do you suggest instead?

Reply:
I think because this lock is only meant to better handle a race condition maybe one second would be enough? What are your thoughts on this @DavidAnderson684? I'm not too familiar with the intricacies of the Semaphore class.

Reply:
Yes, the maximum length of the lock should be related to how long you reasonably expect something to take in the code that is going to run whilst the lock is being held.
i.e. You're saying "let's wait up to (this) amount of time, and after that, we don't expect the code in the other thread that locked this to still be running, and so after that we will try again". So, you don't want to have a time interval that is implausibly long because then you're just wasting time and as Michael says, if it's a lock that other processes may want to have, you're going to hold them all up.
The default is 300 because its initial use was for backups in UpdraftPlus - those may well run for 300 seconds on a large site, and the only thing that's going to call for the lock is a new backup process, and holding those up is the right thing to do. But that's different here.